app/src/main/java/com/eveningoutpost/dexdrip/Services/NsRestApiReader.java

```diff
@@ -0,0 +1,137 @@
+package com.eveningoutpost.dexdrip.Services;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.eveningoutpost.dexdrip.Models.UserError.Log;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import retrofit.Call;
+import retrofit.Response;
+import retrofit.http.GET;
+import retrofit.http.Query;
+import retrofit.Retrofit;
+import retrofit.GsonConverterFactory;
+
+import com.squareup.okhttp.logging.HttpLoggingInterceptor;
+import com.squareup.okhttp.OkHttpClient;
+
+
+public class NsRestApiReader {
+
+	class NightscoutBg {
+	    double xDrip_raw; // raw_data
+	    double xDrip_filtered; // filtered_data;
+	    Long date; // timestamp
+	    double sgv; // calculated_bg
+	}
+
+	class NightscoutMbg {
+	    Long date; // timestamp
+	    double mbg; // calculated_bg
+	    double xDrip_slope;
+	    double xDrip_intercept;
+	    double xDrip_estimate_raw_at_time_of_calibration;
+	    double xDrip_slope_confidence;
+	    double xDrip_sensor_confidence;
+	    double xDrip_raw_timestamp;
+	}
+	
+	
+    private final static String TAG = NsRestApiReader.class.getName();
+
+	
+	private INsRestApi CreateNsMethods(String baseUrl) {
+        Retrofit retrofit;
+
+        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();  
+        // set your desired log level
+        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
+
+        OkHttpClient httpClient = new OkHttpClient();  
```

Comment:
What should the cache hold? Should it hold objects that are received from
the remote server?
If yes, how does it know if they were changed or not?


Thanks
Tzachi


On Wed, Jan 6, 2016 at 4:33 PM, Matthias Granberry <notifications@github.com
> wrote:

> In
> app/src/main/java/com/eveningoutpost/dexdrip/Services/NsRestApiReader.java
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/207#discussion_r48962549>
> :
>
> > +	    double xDrip_sensor_confidence;
> > +	    double xDrip_raw_timestamp;
> > +	}
> > +	
> > +	
> > +    private final static String TAG = NsRestApiReader.class.getName();
> > +
> > +	
> > +	private INsRestApi CreateNsMethods(String baseUrl) {
> > +        Retrofit retrofit;
> > +
> > +        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
> > +        // set your desired log level
> > +        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
> > +
> > +        OkHttpClient httpClient = new OkHttpClient();
>
> Handling gaps and backfills are where that approach fails. If you ever
> decide to add treatments for some reason it would also benefit from a
> modest cache.
>
> Reusing an OkHttpClient will cause new requests to use a shared connection
> pool for any new requests. I believe it maintains 6 connections by default.
> There is a greater benefit when using HTTP/2.0 or SSL and doing multiple
> back-to-back requests. This code probably isn't likely to benefit much, but
> the REST uploader certainly would. Some things like redirects to HTTPS will
> benefit from reused clients and from a persistent cache.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/207/files#r48962549>
> .
>